src/App.tsx

```diff
@@ -1,19 +1,161 @@
+/* eslint-disable no-console */
 import React from 'react';
-import './App.css';
+import './App.scss';
 
 import users from './api/users';
+import todos from './api/todos';
+import { Todo } from './types';
+import { TodoList } from './components/TodoList';
+import { prepareTodos } from './prepareTodos';
 
-const App: React.FC = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+type State = {
+  todos: Todo[];
+  title: string;
+  userId: number;
+  isUserSelected: boolean;
+  isTitleWritten: boolean;
+  tooBigTitle: boolean;
 };
 
+class App extends React.Component<{}, State> {
+  state = {
+    todos,
+    title: '',
+    userId: 0,
+    isUserSelected: true,
+    isTitleWritten: true,
+    tooBigTitle: false,
+  };
+
+  handleOnSubmit = (event: React.SyntheticEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!this.validateSumbitData()) {
+      return;
+    }
+
+    if (this.state.title.length > 60) {
+      return;
+    }
+
+    this.setState(current => {
+      const newTodo:Todo = {
+        userId: current.userId,
+        title: current.title,
+        id: current.todos.length + 1,
```

Comment:
id is a maximal id in the array + 1
What if tomorrow will you add the delete method?